cms/static/cms/js/modules/cms.toolbar.js

```diff
@@ -312,153 +496,179 @@
+                    // add delay to hide if delay > 0
+                    this.timer = setTimeout(function () {
+                        that.closeMessage();
+                    }, delay);
                 }
-                // add delay to hide
-                this.timer = setTimeout(function () {
-                    that.closeMessage();
-                }, delay || this.options.messageDelay);
             },
 
-            closeMessage: function () {
-                this.messages.fadeOut(300);
+            /**
+             * Closes the message window underneath the toolbar.
+             *
+             * @method open
+             * @deprecated this will be moved to `cms.messages.js`
+             */
+            closeMessage: function closeMessage() {
+                this.ui.messages.fadeOut(this.options.toolbarDuration);
                 // unlock toolbar
                 this._lock(false);
             },
 
-            openAjax: function (url, post, text, callback, onSuccess) {
+            /**
+             * Closes the message window underneath the toolbar.
+             *
+             * @method open
+             */
+            openAjax: function (opts) {
                 var that = this;
-
-                // check if we have a confirmation text
+                // url, post, text, callback, onSuccess
+                var url = opts.url;
+                var post = opts.post;
+                var text = opts.text || '';
+                var callback = opts.callback;
+                var onSuccess = opts.onSuccess;
                 var question = (text) ? confirm(text) : true;
+
                 // cancel if question has been denied
                 if (!question) {
                     return false;
                 }
 
                 // set loader
-                this._loader(true);
-
-                $.ajax({
-                    'type': 'POST',
-                    'url': url,
-                    'data': (post) ? JSON.parse(post) : {},
-                    'success': function (response) {
-                        CMS.API.locked = false;
-
-                        if (callback) {
-                            callback(that, response);
-                            that._loader(false);
-                        } else if (onSuccess) {
-                            CMS.API.Helpers.reloadBrowser(onSuccess, false, true);
-                        } else {
-                            // reload
-                            CMS.API.Helpers.reloadBrowser(false, false, true);
-                        }
-                    },
-                    'error': function (jqXHR) {
-                        CMS.API.locked = false;
-                        that.showError(jqXHR.response + ' | ' + jqXHR.status + ' ' + jqXHR.statusText);
+                this.showLoader();
+
+                return $.ajax({
+                    type: 'POST',
+                    url: url,
+                    data: (post) ? JSON.parse(post) : {}
```

Comment:
why do we have this? i though jquery does the parsing for you. so it could just be `data: post`